### PR TITLE
Add faster version of drawcalls example

### PR DESCRIPTION
**Pull request by p5aholic**
I made faster version of [webgl_buffergeometry_drawcalls](https://threejs.org/examples/webgl_buffergeometry_drawcalls.html) using index buffer.

### CODE REVIEW

**Comment by looeee** [comment]
[live version](https://rawgit.com/p5aholic/three.js/5972b810791bc2ade8668e86879f4115066d8ebb/examples/webgl_buffergeometry_drawcalls_indexed.html)

**Comment by looeee** [comment]
Running this and [webgl_buffergeometry_drawcalls](https://threejs.org/examples/webgl_buffergeometry_drawcalls.html), comparing the frame rates I don't see much if any difference.

**Comment by p5aholic** [comment]
@looeee Compare [normal version](https://rawgit.com/p5aholic/three.js/5972b810791bc2ade8668e86879f4115066d8ebb/examples/webgl_buffergeometry_drawcalls.html) and [indexed version](https://rawgit.com/p5aholic/three.js/5972b810791bc2ade8668e86879f4115066d8ebb/examples/webgl_buffergeometry_drawcalls_indexed.html)
I've also changed normal version. See [this change.](https://github.com/mrdoob/three.js/pull/13652/files#diff-6cf0feb52883d371841848da429d23ba)

**Comment by looeee** [comment]
@p5aholic I've tested your versions against each other with a few different settings up to max, and I still can't see any difference in framerate.

**Comment by p5aholic** [comment]
@looeee Really? Are the framerates of the two examples 60? If so, your pc is too high spec to see any difference...

**Comment by looeee** [comment]
>   Are the framerates of the two examples 60? If so, your pc is too high spec to see any difference...

I wish! 😆 
When I put the settings up to max I'm getting around 3 or 4 fps in both.

**Comment by p5aholic** [comment]
@looeee It seems `minDistance` doesn't make any difference. Adjusting only `particleCount` is simple    for comparison.

**Comment by RemusMar** [comment]
> Adjusting only particleCount is simple for comparison.

As I said before, I get the same FPS value in both cases.
I understand what are you trying to do, but it doesn't work here.
In a few words:
the indexed access is always the fastest method, but to get a significant boost you need a very large array.
In this case you need 20-50,000 particles.
But there are 2 big problems is this case:
1) above 5000 particles on many machines you will get a browser crash with the "Out of memory" error message.
2) the global performances will be terrible anyway
cheers

**Comment by looeee** [comment]
> It seems minDistance doesn't make any difference. 

OK, setting min distance to 50 and playing around with the other sliders I'm starting to see a difference, maybe around 20-30% faster with the indexed example.

**Comment by p5aholic** [comment]
@looeee thanks!

**Comment by Mugen87** [comment]
I don't think we need two examples of `webgl_buffergeometry_drawcalls`. One should be sufficient.

File: examples/webgl_buffergeometry_drawcalls_indexed.html [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+				renderer.setSize( window.innerWidth, window.innerHeight );
+
+			}
+
+			function animate() {
+				
+				// update particle position
+				for ( var i = 0; i < particleCount; i++ ) {
+
+					var pData = particlesData[ i ];
```
Comment (fernandojsg) [inline examples/webgl_buffergeometry_drawcalls_indexed.html]: This must be initialized in its own loop as in the original code https://github.com/mrdoob/three.js/blob/dev/examples/webgl_buffergeometry_drawcalls.html#L209-L210
as you will be using querying `particleDataB` indexed by `j > i` without being initialized correctly. https://github.com/mrdoob/three.js/blob/dev/examples/webgl_buffergeometry_drawcalls.html#L237
https://github.com/mrdoob/three.js/blob/dev/examples/webgl_buffergeometry_drawcalls.html#L248

**Comment by fernandojsg** [comment]
Both examples are not working exactly on the same way, if you check the previous example the lines go smoothly appearing and disappearing while on the current PR they don't. I believe it's because of the modified code on the initialization on the piece commented before.
In any case I agree with @Mugen87 that I don't see the point on having two examples with most of the base code being similar. If after changing that piece of code so both examples works on the same way there's still a noticeable difference in performance I would suggest to modify the original example and add another option to the UI to switch between indexed and non-indexed instead of creating a new one.

**Comment by p5aholic** [comment]
@fernandojsg Thanks!
I also noticed lines don't go smoothly appearing and disappearing. But it's because of that one point can affect more than two lines. In my example, alpha value will be overwritten in for loop. I understood my example cannot behave exactly the same as yours. Let me think about it.

**Comment by fernandojsg** [comment]
Yeah that makes sense, it's been a long time since I created that example, maybe I was running on the same issue as you with the alpha value shared across each vertice umm I barely remember thinking about that problem too

**Comment by Mugen87** [comment]
Closing. As discussed before, it makes more sense the improve the existing example than adding a new one.

**Pull request closed without merging**